rse-r-package.Rmd

```diff
@@ -0,0 +1,590 @@
+# R Packaging {#rse-r-package}
+
+## Questions {#rse-r-package-questions}
+
+```{r, child="questions/rse-r-package.md"}
+```
+
+## Objectives {#rse-r-package-objectives}
+
+```{r, child="objectives/rse-r-package.md"}
+```
+
+## What's in an R package? {#rse-r-package-intro}
+
+> Another response of the wizards,
+> when faced with a new and unique situation,
+> was to look through their libraries to see if it had ever happened before.
+> This was...a good survival trait.
+> It meant that in times of danger you spent the day sitting very quietly in a building with very thick walls.
+>
+> -- Terry Pratchett
+
+The more software you write,
+the more you realize that a programming language is really just
+a way to build and combine software libraries.
+Every widely-used language now has an online repository
+from which people can download and install such libraries,
+and sharing yours is a great way to contribute to the community
+that has helped you get where you are.
+This lesson shows you how to use R's tools to do this.
+
+> ### CRAN
+>
+> [CRAN][cran] is the Comprehensive R Archive Network.
+> Like the [Python Package Index][pypi],
+> it is the place to go to find the packages you need.
+> CRAN's rules are famously strict,
+> which ensures that packages run for everyone,
+> but which also makes package development a little more onerous than it might be.
+
+An R package must contain the following files:
+
+-   The text file `DESCRIPTION` (with no suffix) describes what the package does,
+    wrote it,
+    and what other packages it requires to run.
+    We will edit its contents as we go along.
+
+-   `NAMESPACE`,
+    (whose name also has no suffix)
+    contains the names of everything exported from the package
+    (i.e., everything that is visible to the outside world).
+    As we will see,
+    we should leave its management in the hands of RStudio.
+
+-   Just as `.gitignore` tells Git what files in a project to ignore,
+    `.Rbuildignore` tells RStudio which files *shouldn't* be included in the package.
+
+-   All of the R source for our package must go in a directory called `R`;
+    sub-directories below this are not allowed.
+
+-   As you would expect from its name,
+    the `data` directory contains all the data in our package.
+    In order for it to be loadable as part of the package,
+    the data must be saved in R's custom `.rda` format.
+    We will see how to do this below.
+
+-   Manual pages go in the `man` directory.
+    The bad news is that they have to be in a sort-of-LaTeX format
+    that is only a bit less obscure than the runes inscribed on the ancient dagger
+    your colleague brought back from her latest archeological dig.
+    The good news is,
```

Comment:
Other good news is that you can write these documentations in [Markdown](https://cran.r-project.org/web/packages/roxygen2/vignettes/markdown.html)

Reply:
:+1: